### ISSUE TITLE
Question: does skaffold support Minikube with podman?

### ISSUE THREAD

**Issue by stephanwesten** [GoogleContainerTools/skaffold#9194]
Hi,

Since my company moved away from Docker (Desktop), we have settled on Podman. 
My question: Is Podman supported?

Reading the documentation like https://skaffold.dev/docs/builders/builder-types/docker/#dockerfile-with-docker-locally I get the impression that I need to have Docker.

I also tried running Minikube with 'minikube start --driver=podman --container-runtime=cri-o', but when i use skaffold dev commands, I get errors like "the docker-env command only supports the docker and containerd runtimes"

So it seems you need to have Docker running. However the https://skaffold.dev/docs/quickstart/ does not list Docker. Hence my confusion.

**Comment by ericzzzzzzz** [comment]
Hi, @stephanwesten , to use podman you need to do two things 
 - set DOCKER_HOST env variable to your podman socket, please take a look this to see how to do it on different platform, https://podman-desktop.io/docs/migrating-from-docker/using-the-docker_host-environment-variable 
 - pass `--check-cluster-node-platforms=false` flag when running skaffold command , the flag won't be necessary on the next release for using podman : )

**Comment by stephanwesten** [comment]
Ah, I'm using a Mac....

So I should conclude that skaffold does not support Podman on Mac? 

Also in the next release it won't work? (that leaves out quite an audience...)

**Comment by ericzzzzzzz** [comment]
> Ah, I'm using a Mac....
> 
> So I should conclude that skaffold does not support Podman on Mac?
> 
> Also in the next release it won't work? (that leaves out quite an audience...)

sorry for the confusion ,  You can use `--check-cluster-node-platforms=false` for any OS. However, to make podman work without that flag, that requires a bit extra work, without flag it will only work on linux  for the next release.

**Comment by stephanwesten** [comment]
Unfortunately I don't even make it to the scaffold command...

I tried this:

➜  skaffoldtest minikube start --driver=podman --container-runtime=cri-o --profile custom
😄  [custom] minikube v1.32.0 on Darwin 13.6.2 (arm64)
✨  Using the podman (experimental) driver based on existing profile

⛔  Requested memory allocation (1888MB) is less than the recommended minimum 1900MB. Deployments may fail.

👍  Starting control plane node custom in cluster custom
🚜  Pulling base image ...
E1129 10:42:12.582696     733 cache.go:189] Error downloading kic artifacts:  not yet implemented, see issue #8426
🏃  Updating the running podman "custom" container ...
🎁  Preparing Kubernetes v1.28.3 on CRI-O 1.24.6 ...
🔗  Configuring CNI (Container Networking Interface) ...
🔎  Verifying Kubernetes components...
    ▪ Using image gcr.io/k8s-minikube/storage-provisioner:v5
🌟  Enabled addons: storage-provisioner, default-storageclass
🏄  Done! kubectl is now configured to use "custom" cluster and "default" namespace by default
➜  skaffoldtest 
➜  skaffoldtest eval $(minikube -p custom docker-env)
❌  Exiting due to MK_USAGE: the docker-env command only supports the docker and containerd runtimes


Although you mentioned not needed, I did set the DOCKER_HOST:

➜  skaffoldtest echo $DOCKER_HOST       
unix:///Users/stephan.westen/.local/share/containers/podman/machine/qemu/podman.sock

**Comment by ericzzzzzzz** [comment]
yeah. Skaffold uses `minikube dokcer-env` to get builder environment variables, that command only works with when minikube cluster is configured with `docker runtime` ,  is it possible for you to use docker as container-runtime on minikube ?

**Comment by stephanwesten** [comment]
I could. 

As more people will follow the same path as I tried, I suggest to update the docs.

Thanks for the help!

**Comment by afbjorklund** [comment]
Minikube supports multiple container runtimes, but skaffold is hardcoded to only use docker...

* https://github.com/GoogleContainerTools/skaffold/issues/5739

Therefore, we need to add legacy compatibility layers to minikube for cri-o and for containerd.

* https://github.com/kubernetes/minikube/issues/15389

**Comment by initharrington** [comment]
doesn't seem I am able to use skaffold in its current state or with a lot of hacking on it.  Company doesn't allow docker for desktop.

Podman/Minikube get the same error as above:

```
invalid skaffold config: getting minikube env: running [/opt/homebrew/bin/minikube docker-env --shell none -p minikube --user=skaffold]
 - stdout: "false exit code 14\n"
 - stderr: "X Exiting due to MK_USAGE: the docker-env command only supports the containerd runtime with the docker driver\n"
 - cause: exit status 14
```

**Comment by afbjorklund** [comment]
I think that docker-env has been added for containerd, but not yet available for  cri-o (podman)

Most likely it also needs a newer version of podman, instead of the old 3.x currently available?
